datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -824,6 +854,30 @@ mod tests {
             Operator::Lt,
             DataType::Date64
         );
+        test_coercion_binary_rule!(
```

Comment:
👍  nice

I double checked that this casting is supported in arrow-rs, which they are ✅ 

https://github.com/apache/arrow-rs/blob/e44cb5b478257751916d1674292123eaf5d80a7c/arrow-cast/src/cast.rs#L149-L158

Reply:
Thank you very much for checking this, @alamb!